File: docs/research/domain-foundations/code/prim-text/make_fixtures.mjs
import { writeFileSync, mkdirSync } from "node:fs";
import { dirname, join } from "node:path";
const root = dirname(new URL(import.meta.url).pathname);
const fixtures = join(root, "fixtures");
mkdirSync(fixtures, { recursive: true });
const seed = "Hello café κόσμε नमस्ते سلام — Jet tokenizer search. The quick brown fox jumps over 42 logs; user_id=31415 level=INFO.\n";
const seedBytes = Buffer.from(seed, "utf8");
const corpus = Buffer.alloc(1_048_576);
for (let at = 0; at < corpus.length; at += seedBytes.length) {
  seedBytes.copy(corpus, at, 0, Math.min(seedBytes.length, corpus.length - at));
}
writeFileSync(join(fixtures, "corpus_1m.txt"), corpus);
const logSeed = Buffer.from(
  "2026-09-02T00:00:00Z INFO api request user_id=31415 path=/orders status=200\n" +
  "2026-09-02T00:00:01Z WARN api retry user_id=27182 path=/orders status=503\n" +
  "2026-09-02T00:00:02Z ERROR db timeout user_id=16180 path=/orders status=500\n",
  "ascii",
);
const logs = Buffer.alloc(100_000_000);
for (let at = 0; at < logs.length; at += logSeed.length) {
  logSeed.copy(logs, at, 0, Math.min(logSeed.length, logs.length - at));
}
writeFileSync(join(fixtures, "logs_100m.txt"), logs);
console.log(`corpus_1m_bytes=${corpus.length}`);
console.log(`logs_100m_bytes=${logs.length}`);

File: docs/research/domain-foundations/code/prim-text/make_fixtures.py
from pathlib import Path

root = Path(__file__).parent
fixtures = root / "fixtures"
fixtures.mkdir(parents=True, exist_ok=True)

# Repeated multilingual text makes both Unicode segmentation and BPE pairs observable.
seed = (
    "Hello café κόσμε नमस्ते سلام — Jet tokenizer search. "
    "The quick brown fox jumps over 42 logs; user_id=31415 level=INFO.\n"
)
# Exactly 1 MiB of valid UTF-8 corpus for tokenizer training.
corpus = (seed * ((1_048_576 // len(seed.encode("utf-8"))) + 1)).encode("utf-8")[:1_048_576]
(fixtures / "corpus_1m.txt").write_bytes(corpus)

# Exactly 100,000,000 bytes of newline-delimited logs. The final line is complete.
log_seed = (
    "2026-09-02T00:00:00Z INFO api request user_id=31415 path=/orders status=200\n"
    "2026-09-02T00:00:01Z WARN api retry user_id=27182 path=/orders status=503\n"
    "2026-09-02T00:00:02Z ERROR db timeout user_id=16180 path=/orders status=500\n"
)
need = 100_000_000
out = bytearray()
while len(out) + len(log_seed.encode("ascii")) <= need:
    out.extend(log_seed.encode("ascii"))
if len(out) < need:
    out.extend(b"X" * (need - len(out)))
(fixtures / "logs_100m.txt").write_bytes(out)
print(f"corpus_1m_bytes={len(corpus)}")
print(f"logs_100m_bytes={len(out)}")
